build.py

```diff
@@ -157,17 +157,24 @@ def process_index(out_dir):
 
 def print_errors(errors, js_files):
   for error in errors:
-    if error['file'].lower().find('externs') >= 0:
-      filename = error['file']
-    else:
-      fileno = int(error['file'][6:]) - 1
+    filename = error.get('file', '')
+    if not filename:
+      filename = missing_key_string('file')
+    elif filename.lower().find('externs') < 0:
+      fileno = int(filename[6:]) - 1
       filename = js_files[fileno]
     if 'error' in error:
       text = error['error']
-    else:
+    elif 'warning' in error:
       text = error['warning']
-    print(filename + ':' + str(error['lineno']) + ' ' + text)
-    print(error['line'])
+    else:
+      text = missing_key_string('error/warning')
+    print(filename + ':' + str(error.get('lineno', missing_key_string('lineno'))) + ' ' + text)
+    print (error.get('line', missing_key_string('line')))
+
+
+def missing_key_string(key):
```

Comment:
This function's name makes it hard to read. I see that

`missing_key_string('foo')`

will return

`['foo' key missing]`

but it seems like a weird name. Maybe something like `get_missing_key_msg` makes more sense?

Reply:
Done